hack/generator/pkg/astmodel/struct_type.go

```diff
@@ -80,18 +80,16 @@ func (structType *StructType) RequiredImports() []*PackageReference {
 	return result
 }
 
-// References this type has to the given type
-func (structType *StructType) References(d *TypeName) bool {
-
+// References returns the combination of all the types the fields refer to.
+func (structType *StructType) References() TypeNameSet {
+	var results TypeNameSet
 	for _, field := range structType.fields {
-		if field.FieldType().References(d) {
-			return true
+		for ref := range field.FieldType().References() {
+			results = results.Add(ref)
```

Comment:
Since `References()` returns a `TypeNameSet`, you could make this just

```
var results TypeNameSet
for _, field := range structType.fields {
    results = results.Union(field.FieldType().References()
}
```

Reply:
I didn't do this because it ends up being quadratic, although it's probably not a big deal in the structs we deal with. (Although it wouldn't be if I just made Unions an asymmetric method, hmmm.)